pkg/cmd/cli/cmd/edit.go

```diff
@@ -187,23 +199,30 @@ func (o *EditOptions) RunEdit() error {
 		if err != nil {
 			return preservedFile(err, file, o.out)
 		}
-		if bytes.Equal(original, edited) {
+		// Compare content without comments
+		uncommentedOriginal, originalErr := yaml.ToJSON(original)
+		uncommentedEdited, editedErr := yaml.ToJSON(edited)
```

Comment:
It's fine, although it would be better to express this as a separate package-private method.